js/modules/k6/html/html_test.go

```diff
@@ -425,10 +442,41 @@ func TestParseHTML(t *testing.T) {
 		t.Run("Map with attr must return string", func(t *testing.T) {
 			v, err := rt.RunString(`doc.find("#select_multi").map(function(idx, val) { return val.attr("name") })`)
 			if assert.NoError(t, err) {
-				mapped, ok := v.Export().([]string)
+				mapped, ok := v.Export().([]goja.Value)
+				assert.True(t, ok)
+				assert.Equal(t, 1, len(mapped))
+				assert.Equal(t, "select_multi", mapped[0].String())
+			}
+		})
+		t.Run("Valid XML", func(t *testing.T) {
+			rt := getTestRuntimeWithDoc(t, testXML)
+			testScript := `
+			doc
+				.find('Buckets')
+				.children()
+				.map(function (idx, bucket) {
+					let bucketObj = {}
+					bucket.children().each(function (idx, elem) {
+						switch (elem.nodeName()) {
+							case 'name':
+								Object.assign(bucketObj, { name: elem.textContent() })
+								break
+							case 'creationdate':
+								Object.assign(bucketObj, { creationDate: elem.textContent() })
+								break
+						}
+					})
+				return bucketObj
+				})
+			`
+
+			v, err := rt.RunString(testScript)
+			if assert.NoError(t, err) {
+				mapped, ok := v.Export().([]goja.Value)
 				assert.True(t, ok)
 				assert.Equal(t, 1, len(mapped))
-				assert.Equal(t, []string{"select_multi"}, mapped)
+				assert.Equal(t, "myBucketName", mapped[0].ToObject(rt).Get("name").String())
+				assert.Equal(t, "myCreationDateTimestamp", mapped[0].ToObject(rt).Get("creationDate").String())
 			}
```

Comment:
I would prefer if the asserts on this are in the script. 

My idea with there being a bigger a script is to check that we don't break how it work for the users of it in js, not something else.

Reply:
Fair enough 👍🏻 
How would you approach asserting in the script itself in the context of testing, should I use `check`?

Reply:
As we do it everywhere else - use `if` to check and `throw` to report that something isn't okay

Reply:
Thanks for the heads-up 🙇🏻  willco 👍🏻

Reply:
I adapted the test to use the `if` and `throw` directly in the script instead: https://github.com/grafana/k6/pull/2534/files#diff-863787185e2a4e0fc11a15c2a0038772a31f6b54e71b964107c7a1af5ef26362R472. Does that fit what you had in mind?